src/gpb_parse.yrl

```diff
@@ -410,6 +412,9 @@ flatten_fields(FieldsOrDefs, FullName) ->
                             {[F | Fs], Ds};
                        (#gpb_oneof{}=O, {Fs,Ds}) ->
                             {[O | Fs], Ds};
+                       ({{extend, Msg},Exts}, {Fs,Ds}) ->
+                            QExt = {{extend, ['.',Msg]}, Exts},
```

Comment:
This is the manual qualification that I'm not comfortable with. If this `extend` was happening inside a message with a `FullName = ['.','some','.','package','.','m1']` I think it should have a qualified name of `['.', 'some', '.', 'package', '.', Msg]`